File: FastFiz/perl/fftest.pl
#!/usr/bin/perl -w
use Pool;
use Data::Dumper;
use Pool::Rules;

print Pool::Fiz::getFastFizVersion()."\n";
print Pool::Rules::getRulesVersion()."\n";
my $state;
while (<>) {
#  print;
  if (m{TSTATE (.+)$}o) {
    my $tsstring=$1;
    $state=new Pool::Fiz::TableState();
    $state->fromString($tsstring);
  } elsif (m{TSHOT ([-\.\d]+) ([-\.\d]+) ([-\.\d]+) ([-\.\d]+) ([-\.\d]+)$}o) {
    my $sp = new Pool::Fiz::ShotParams($1,$2,$3,$4,$5);
    my $shot=$state->executeShot($sp,1,1);
    print "\n\n";
    print $state->toString();
    print "\n\n";
    my @evlist=$shot->getEventList();
    print Dumper([map {$_->toString()} @evlist]);
  }
}

File: FastFiz/python/fftest.py
#!/usr/bin/python
import FastFiz
import Rules
import sys

#print(FastFiz.getFastFizVersion())
shotParams = FastFiz.ShotParams()
shotParams.v = 4
shotParams.a = 6
shotParams.b = 8
shotParams.phi = 10
shotParams.theta = 11

gameShot = Rules.GameShot()
gameShot.params = shotParams
gameShot.cue_x = 1
gameShot.cue_y = 2

gameState = Rules.GameState.RackedState(Rules.GT_EIGHTBALL)
gameState.executeShot(gameShot)

print(shotParams)
print(Rules.getRulesVersion())

#print("Comparing two initial game States")
#print(gameState.toString())
gameState = Rules.GameState.RackedState(Rules.GT_EIGHTBALL)
gameStateString = gameState.toString()
#print(gameStateString)
newGameState = Rules.GameState.Factory(gameStateString)
newGameStateString = newGameState.toString()
#print(newGameStateString)

print("Onto tests version 2")
print("Without the Eight-Ball Specific Stuff Appended")
constructorString = "1 4 0 0 1 1 1 0.028575 1 0 0.558 1.69935"
constructorString = "1 4 0 0 1 1 0.028575 1 0 0.558 1.69935"
newerGameState = Rules.GameState.Factory(constructorString)
print(constructorString)
print(newerGameState.toString())
print("With the Eight-Ball Specific Stuff Appended")
constructorString = "1 4 0 0 1 1 1 0.028575 1 0 0.558 1.69935 0 1"
constructorString = "1 4 0 0 1 1 0.028575 1 0 0.558 1.69935 0 1"
newerGameState = Rules.GameState.Factory(constructorString)
print(constructorString)
print(newerGameState.toString())
print(newerGameState.isOpenTable())
print(newerGameState.playingSolids())


sys.settrace(gameState)

